pinecone/client.go

```diff
@@ -6,17 +6,22 @@ import (
 	"fmt"
 	"github.com/deepmap/oapi-codegen/v2/pkg/securityprovider"
 	"github.com/pinecone-io/go-pinecone/internal/gen/control"
+	"github.com/pinecone-io/go-pinecone/internal/provider"
+	"github.com/pinecone-io/go-pinecone/internal/useragent"
 	"io"
 	"net/http"
 )
 
 type Client struct {
 	apiKey     string
 	restClient *control.Client
+	sourceTag  string
 }
 
 type NewClientParams struct {
 	ApiKey string
+	// optional fields
+	SourceTag string
```

Comment:
```suggestion
	SourceTag string // optional
```

Reply:
unless there's a perf reason (which i don't think there is) we probably want to avoid pointers here as they tend to confuse newbies